api/src/main/java/org/openmrs/module/ModuleUtil.java

```diff
@@ -62,11 +62,17 @@ private ModuleUtil() {
 	
 	private static final Logger log = LoggerFactory.getLogger(ModuleUtil.class);
 	
+	// adapted from semver.org
+		private static final Pattern VERSION_PATTERN = Pattern.compile(
+			""^(?<major>0|[1-9]\\d*)\\.(?<minor>0|[1-9]\\d*)\\.(?<patch>0|[1-9]\\d*)(?:-(?<snapshot>SNAPSHOT))?$""
```

Comment:
```suggestion
			"^(?<major>0|[1-9]\\d*)\\.(?<minor>0|[1-9]\\d*)\\.(?<patch>0|[1-9]\\d*)(?:-(?<snapshot>SNAPSHOT))?$"
```

You've got two sets of double quotes there, which doesn't work very well.